medicines/doc-index-updater/tests/support/document_api.rs

```diff
@@ -1,15 +1,38 @@
 use super::get_ok;
-use doc_index_updater::models::{Document, DocumentType, FileSource, JobStatus, JobStatusResponse};
+use reqwest::RequestBuilder;
+
+use doc_index_updater::{
+    get_env,
+    models::{Document, DocumentType, FileSource, JobStatus, JobStatusResponse},
+};
 use reqwest::Error;
 use uuid::Uuid;
 
+pub fn get_basic_auth_credentials() -> (String, String) {
+    let username: String = get_env("BASIC_AUTH_USERNAME").unwrap();
+    let password: String = get_env("BASIC_AUTH_PASSWORD").unwrap();
+
+    (username, password)
+}
+
+pub trait BasicAuthFromEnv {
+    fn with_auth(self) -> RequestBuilder;
+}
+
+impl BasicAuthFromEnv for RequestBuilder {
+    fn with_auth(self) -> RequestBuilder {
+        let (username, password) = get_basic_auth_credentials();
```

Comment:
if you did want this trait to imply basic auth, it could be `WithBasicAuth / with_basic_auth(self, String, String)`, with `username` and `password` passed as arguments

Reply:
It's more like `WithBasicAuthFromEnvironmentVars` and `with_basic_auth_from_environment_vars`